department-of-veterans-affairs/caseflow
# Death Dismissal Part 5 Task Actions

Resolves #12211

Finishes work started in #12529; #12535; #12543;  #12544

:red_circle: MERGE BLOCKED by #12535 & #12543 &  #12544. Contains redundant code until those are merged. 
:red_circle:  MERGE BLOCKED by @laurjpeterson  pending convo with stakeholders (or feature flag?)

### Description
Adds the Task Action Death Dismissal, allowing the VLJ Support Admin user to do a death dismissal on the case.

### Acceptance Criteria
- [ ] A VLJ Support admin can perform a death dismissal on a LegacyAppeal assigned to VLJ Support when the vacols case has a Final Notice of Death

### Testing Plan
1. Sign in as a Colocated Admin & navigate to a legacy appeal
2. Mark that LegacyAppeal with a SFNOD on the console
    `appeal.case_record.correspondent.update!(sfnod: 4.days.ago)`
3. Confirm you have the option `Death Dismissal` in the drop down
4. Confirm the action pops up a confirmation modal
5. Submitting the death dismissal results in a success message
6. On the console, confirm the appeal location is now 66 & the appeal tasks are all cancelled.


A part of a larger set of PRs.
#12529 #12535 #12543 #12544 `#12545`
